### PR TITLE
Add Grafana integration to OAuth2 documentation

### PR DESCRIPTION
**Pull request by pdostal**
Fixes #
- Adds Grafana integration to OAuth2 documentation
  Significant part of the snippet is from @matthewpi
  As promised in #2597

Checklist

- [ ] This pr contains no AI generated code
- [ ] cargo fmt has been run
- [ ] cargo clippy has been run
- [ ] cargo test has been run and passes
- [x] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

File: book/src/integrations/oauth2.md [hunk 1]
```diff
@@ -483,3 +483,65 @@ The `email` scope needs to be passed and thus the mail attribute needs to exist
...
+$ kanidm system oauth2 update-claim-map-join 'grafana' 'grafana_role' array
+$ kanidm system oauth2 update-claim-map 'grafana' 'grafana_role' 'grafana_superadmins' 'GrafanaAdmin'
+$ kanidm system oauth2 update-claim-map 'grafana' 'grafana_role' 'grafana_admins' 'Admin'
+$ kanidm system oauth2 update-claim-map 'grafana' 'grafana_role' 'grafana_editors' 'Editor'
+```
+
+Don't forget that every Grafana user needs be member of one of above group and have name and e-mail:
+
+```bash
+$ kanidm person update pdostal --legalname "Personal Name" --mail "user@example.com"
```
Comment (yaleman) [inline book/src/integrations/oauth2.md:523]:
```suggestion
$ kanidm person update user --legalname "Personal Name" --mail "user@example.com"
```

File: book/src/integrations/oauth2.md [hunk 2]
```diff
@@ -483,3 +483,65 @@ The `email` scope needs to be passed and thus the mail attribute needs to exist
...
+And add the following to your Grafana config:
+
+```toml
+[auth.generic_oauth]
+enabled = true
+name = Kanidm
+client_id = grafana
+client_secret = <SECRET>
+scopes = openid,profile,email,groups
+auth_url = https://idm.pdostal.cz/ui/oauth2
```
Comment (yaleman) [inline book/src/integrations/oauth2.md:536]: All domain names should be idm.example.com:

```suggestion
auth_url = https://idm.example.com/ui/oauth2
```

**Review by yaleman — CHANGES_REQUESTED** [review]
For actually-valid TOML, the strings should be wrapped in double quotes.

File: book/src/integrations/oauth2.md [hunk 3]
```diff
@@ -483,3 +483,65 @@ The `email` scope needs to be passed and thus the mail attribute needs to exist
...
+token_url = https://idm.pdostal.cz/oauth2/token
+api_url = https://idm.pdostal.cz/oauth2/openid/grafana/userinfo
+use_pkce = true
+use_refresh_token = true
+allow_sign_up = true
+login_attribute_path = prefered_username
```
Comment (tacerus) [inline book/src/integrations/oauth2.md:542]: Note my comment in https://github.com/kanidm/kanidm/issues/2597#issuecomment-2016511924.

Comment (pdostal) [reply book/src/integrations/oauth2.md:542]: Saw it and then forgot about it. Thank you.

**Comment by pdostal** [comment]
All suggestions applied, thank you

**Review by yaleman — CHANGES_REQUESTED** [review]
Please ensure it's valid TOML (quote strings) - note the red error bars in the github diff.

**Comment by matthewpi** [comment]
Grafana's config format is an `ini`, not `toml`.

**Comment by tacerus** [comment]
The issue is in line 529 where a wrong markdown syntax hint is used.

**Comment by yaleman** [comment]
Thanks @tacerus, yep, that's why I was flagging it. If it's not TOML then don't mark it as such please 😄

**Review by yaleman — APPROVED** [review]
Thanks!

File: book/src/integrations/oauth2.md [hunk 4]
```diff
@@ -483,3 +483,65 @@ The `email` scope needs to be passed and thus the mail attribute needs to exist
...
+
+### Grafana
+
+Grafana is a open source analytics and interactive visualization web application. It provides charts, graphs, and alerts when connected to supported data source.
+
+Prepare the environment:
+
+```bash
+$ kanidm system oauth2 create grafana "grafana.domain.name" https://grafana.domain.name
+$ kanidm system oauth2 update-scope-map grafana grafana_users email openid profile
```
Comment (mweinelt) [inline book/src/integrations/oauth2.md:495]: Shouldn't this include the groups scope, as it gets requested by Grafana with the config below?

Comment (yaleman) [reply book/src/integrations/oauth2.md:495]: Yep, that'd make sense.

**Pull request merged**